core/modules/parsers/wikiparser/wikiparser.js

```diff
@@ -165,6 +165,21 @@ WikiParser.prototype.findNextMatch = function(rules,startPos) {
 	return matchingRule;
 };
 
+WikiParser.prototype.parseRule = function(rule) {
+	var start = this.pos,
+		blocks = rule.parse();
+	for(var i=0; i<blocks.length; i++) {
```

Comment:
TW core code most of the times uses a construction like: 

```
var i, length  // var defs outside for

for(i=0, length=blocks.length; i<length; i++) {
```

Reply:
Apologies @pmario that's not quite right, the core does frequently declare variables within a "for" statement.

Reply:
Just a FYI: caching the `length` when iterating over an array with a for-loop actually results in worse performance (though probably not measurably). Read http://mrale.ph/blog/2014/12/24/array-length-caching.html for details if you want to know more of what happens under the hood.

TL;DR; of the linked post: When accessing `blocks[i]` inside the loop, the engine has to do a bounds check to make sure `i` is not past the end of `blocks`. When you do not cache the length, but use `i < blocks.length` in the condition, the engine can optimize the bounds check away. If you cache, it cannot (or at least couldn't when the article was written). Don't do premature optimizations like that without measuring first.